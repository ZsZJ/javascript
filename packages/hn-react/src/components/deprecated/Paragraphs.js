import React from 'react';
import PropTypes from 'prop-types';
import EntityListMapper from '../EntityListMapper';
import deprecate from 'util-deprecate';

function Paragraphs({ mapper, paragraphs, page, Wrapper, paragraphProps }) {
  return (
    <EntityListMapper
      entities={paragraphs}
      mapper={mapper}
      entityProps={{ page, ...paragraphProps }}
      {...(Wrapper ? { entityWrapper: Wrapper } : {})}
    />
  );
}

Paragraphs.propTypes = {
  Wrapper: PropTypes.oneOfType([PropTypes.string, PropTypes.func]),
  mapper: PropTypes.oneOfType([PropTypes.shape(), PropTypes.func]).isRequired,
  page: PropTypes.shape(),
  paragraphProps: PropTypes.shape(),
  paragraphs: PropTypes.arrayOf(PropTypes.shape()),
};

Paragraphs.defaultProps = {
  Wrapper: undefined,
  page: undefined,
  paragraphProps: {},
};

export default deprecate(
  Paragraphs,
  'The Paragraphs component is deprecated, use the EntityListMapper component instead',
);
