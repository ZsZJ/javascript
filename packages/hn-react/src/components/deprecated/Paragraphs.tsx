import PropTypes from 'prop-types';
import React from 'react';
import deprecate from 'util-deprecate';
import EntityListMapper from '../EntityListMapper';

const Paragraphs: React.StatelessComponent<ParagraphsProps> = ({
  mapper,
  paragraphs,
  page,
  Wrapper,
  paragraphProps,
}) => (
  <EntityListMapper
    entities={paragraphs}
    mapper={mapper}
    entityProps={{ page, ...paragraphProps }}
    {...(Wrapper ? { entityWrapper: Wrapper } : {})}
  />
);

interface ParagraphsProps {
  Wrapper?: any;
  mapper: any;
  page?: any;
  paragraphProps?: any;
  paragraphs?: any;
}

Paragraphs.propTypes = {
  Wrapper: PropTypes.oneOfType([PropTypes.string, PropTypes.func]),
  mapper: PropTypes.oneOfType([PropTypes.shape({}), PropTypes.func]).isRequired,
  page: PropTypes.shape({}),
  paragraphProps: PropTypes.shape({}),
  paragraphs: PropTypes.arrayOf(PropTypes.shape({})),
};

Paragraphs.defaultProps = {
  Wrapper: undefined,
  page: undefined,
  paragraphProps: {},
};

export default deprecate(
  Paragraphs,
  'The Paragraphs component is deprecated, use the EntityListMapper component instead',
);
