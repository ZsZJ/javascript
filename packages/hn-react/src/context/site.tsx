import { Site } from 'hn';
import React, { createContext, ReactNode } from 'react';
import globalSite from '../utils/site';

// Create a new context.
const { Provider, Consumer } = createContext(globalSite);

// The site provider is the same as the 'Provider', but changes 'site' to
// 'value'.
export const SiteProvider = ({
  site,
  children,
}: {
  site: Site;
  children: ReactNode;
}) => <Provider value={site} children={children} />;

// The consumer is exported as both Site and SiteConsumer. The SiteConsumer can
// be used if also the Site from 'hn' is imported, so the names don't overlap.
export { Consumer as Site, Consumer as SiteConsumer };

// There is also a HOC withSite that adds the site to the props of the
// component. Source of the HOC for Typescript:
// https://stackoverflow.com/a/50613946/1907875
export interface InjectedSiteProps {
  site: Site;
}

export function withSite<P extends InjectedSiteProps>(
  Component: React.ComponentType<P>,
) {
  return function ComponentWithSite(
    props: Pick<P, Exclude<keyof P, keyof InjectedSiteProps>>,
  ) {
    return (
      <Consumer>{theme => <Component {...props} site={theme} />}</Consumer>
    );
  };
}
