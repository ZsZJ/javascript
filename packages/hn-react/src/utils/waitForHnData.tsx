import * as PropTypes from 'prop-types';
import * as React from 'react';

const asyncBootstrapper = require('react-async-bootstrapper'); // tslint:disable-line:no-var-requires

export default async function(children) {
  const context = {
    state: {
      drupalPage: null,
      entities: [],
    },
  };

  class DrupalPageContextProvider extends React.Component {
    static childContextTypes = {
      hnContext: PropTypes.object,
    };

    getChildContext() {
      return { hnContext: context };
    }

    render() {
      return React.Children.only(this.props.children);
    }
  }

  const drupalContextProvider = (
    <DrupalPageContextProvider>{children}</DrupalPageContextProvider>
  );

  await asyncBootstrapper(drupalContextProvider, {
    componentWillUnmount: true,
  });

  return drupalContextProvider;
}
