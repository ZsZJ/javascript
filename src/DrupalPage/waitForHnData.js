import React, { Component } from 'react';
import asyncBootstrapper from 'react-async-bootstrapper'
import PropTypes from 'prop-types';

export default async function (children) {

  const context = {};

  class DrupalPageContextProvider extends Component {
    getChildContext() {
      return {hnContext: context};
    }

    render() {
      return React.Children.only(this.props.children);
    }
  }

  DrupalPageContextProvider.childContextTypes = {
    hnContext: PropTypes.object,
  };

  const drupalContextProvider = (
    <DrupalPageContextProvider>
      {children}
    </DrupalPageContextProvider>
  );

  await asyncBootstrapper(drupalContextProvider);

  return drupalContextProvider;

};
