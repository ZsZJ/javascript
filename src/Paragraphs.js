import React from 'react';
import PropTypes from 'prop-types';
import Paragraph from './Paragraph';

const Paragraphs = ({ mapper, paragraphs, page, Wrapper }) =>
  paragraphs.map((uuid, index) => {
    const paragraph = (
      <Paragraph
        key={uuid.target_uuid}
        uuid={uuid.target_uuid}
        page={page}
        index={index}
        mapper={mapper}
      />
    );
    if(Wrapper) {
      return (
        <Wrapper>
          {paragraph}
        </Wrapper>
      );
    }
    return paragraph;
  });

Paragraphs.propTypes = {
  mapper: PropTypes.oneOfType([
    PropTypes.shape(),
    PropTypes.func,
  ]).isRequired,
  paragraphs: PropTypes.arrayOf(PropTypes.shape()),
  page: PropTypes.shape(),
  Wrapper: PropTypes.oneOfType([
    PropTypes.bool,
    PropTypes.node,
  ]),
};

Paragraphs.defaultProps = {
  Wrapper: false,
};

export default Paragraphs;
