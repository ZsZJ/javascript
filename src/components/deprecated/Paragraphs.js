import React from 'react';
import PropTypes from 'prop-types';
import EntityListMapper from '../../EntityListMapper';
import deprecate from "util-deprecate";

function Paragraphs({ mapper, paragraphs, page, Wrapper, paragraphProps }) {
  return <EntityListMapper
    entities={paragraphs}
    mapper={mapper}
    entityProps={{page, ...paragraphProps}}
    {...(Wrapper ? {entityWrapper: Wrapper} : {})}
  />;
}

Paragraphs.propTypes = {
  mapper: PropTypes.oneOfType([
    PropTypes.shape(),
    PropTypes.func,
  ]).isRequired,
  paragraphs: PropTypes.arrayOf(PropTypes.shape()),
  page: PropTypes.shape(),
  Wrapper: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.func,
  ]),
  paragraphProps: PropTypes.shape(),
};

Paragraphs.defaultProps = {
  Wrapper: undefined,
  paragraphProps: {},
  page: undefined,
};

export default deprecate(Paragraphs, 'The Paragraphs component is deprecated, use the EntityListMapper component instead');
