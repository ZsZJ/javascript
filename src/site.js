import { Site } from 'hn';
import getNested from 'get-nested';

const site = new Site({ url: '' });

if(getNested(() => window.__NEXT_DATA__.HN_DATA)) {
  site.data = window.__NEXT_DATA__.HN_DATA;
  site.url = window.__NEXT_DATA__.HN_URL || site.url;
  site.pagesLoading = {
    [window.location.pathname]: Promise.resolve(),
  };
}

export default site;
