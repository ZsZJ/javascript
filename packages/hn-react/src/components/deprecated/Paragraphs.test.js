import React from 'react';
import renderer from 'react-test-renderer';
import Paragraphs from './Paragraphs';
import site from '../../utils/site';
import waitForHnData from '../../utils/waitForHnData';
import { mapper, uuid } from '../../utils/tests';

jest.mock('../../utils/site', () => {
  return require('../../utils/tests').mockSite();
});

jest.mock('util-deprecate', () => jest.fn(func => func));
console.log = console.warn = console.error = message => {
  throw new Error(message);
};

beforeEach(() => {
  site.getData.mockRestore();
});

describe('Paragraphs', async () => {
  test('with required props', async () => {
    const component = await waitForHnData(
      <Paragraphs mapper={mapper} paragraphs={[{ target_uuid: uuid }]} />,
    );

    expect(renderer.create(component).toJSON()).toMatchSnapshot();
  });

  test('with all props', async () => {
    const component = await waitForHnData(
      <Paragraphs
        mapper={mapper}
        paragraphProps={{
          testProp: 'testPropValue',
        }}
        paragraphs={[{ target_uuid: uuid }]}
        Wrapper={'section'}
        page={{ pageTest: true }}
      />,
    );

    expect(renderer.create(component).toJSON()).toMatchSnapshot();
  });
});
