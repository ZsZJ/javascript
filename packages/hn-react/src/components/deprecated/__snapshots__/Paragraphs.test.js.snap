// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paragraphs with all props 1`] = `
<section>
  <div
    bundle="unique_type_1__unique_bundle_1"
    className="MyCustomMapperComponent"
    entity={
      Object {
        "__hn": Object {
          "entity": Object {
            "bundle": "unique_bundle_1",
            "type": "unique_type_1",
          },
        },
      }
    }
    index={0}
    page={
      Object {
        "pageTest": true,
      }
    }
    paragraph={
      Object {
        "__hn": Object {
          "entity": Object {
            "bundle": "unique_bundle_1",
            "type": "unique_type_1",
          },
        },
      }
    }
    testProp="testPropValue"
  />
</section>
`;

exports[`Paragraphs with required props 1`] = `
<div
  bundle="unique_type_1__unique_bundle_1"
  className="MyCustomMapperComponent"
  entity={
    Object {
      "__hn": Object {
        "entity": Object {
          "bundle": "unique_bundle_1",
          "type": "unique_type_1",
        },
      },
    }
  }
  index={0}
  page={undefined}
  paragraph={
    Object {
      "__hn": Object {
        "entity": Object {
          "bundle": "unique_bundle_1",
          "type": "unique_type_1",
        },
      },
    }
  }
/>
`;
